### ISSUE TITLE
Feature Request: Allow Embedding of elevation and profile REST API

### ISSUE THREAD

**Issue by cymed** [qwc-services/qwc-elevation-service#11]
As an addition to referring to a raster in elevation_dataset, it would be practical to be able to access elevation and profile data in the elevation config through REST APIs by extending the qwc-elevation-service.json

    "config": {
      "title": "Config options",
      "type": "object",
      "properties": {
        "elevation_dataset": {
          "description": "Elevation dataset (file or URL). Example: https://data.sourcepole.com/srtm_1km_3857.tif",
          "type": "string"
        },
        "height_dataset": {
          "description": "Height dataset (file or URL). Example: https://api3.geo.admin.ch/rest/services/height",
          "type": "string"
        },
        "height_epsg": {
          "description": "EPSG of Height dataset (file or URL). Example: 4326",
          "type": "integer"
        },
        "profile_dataset": {
          "description": "Elevation dataset (file or URL). Example: https://api3.geo.admin.ch/rest/services/profile.json",
          "type": "string"
        },
        "profile_epsg": {
          "description": "EPSG of Profile dataset . Example: 4326",
          "type": "integer"
        },
      },
     "oneOf": [
       {
          "required": [
          "elevation_dataset"
          ]
      },
      {
         "required": [
           "height_dataset",
           "height_epsg",
           "profile_dataset",
           "profile_epsg"
         ]
       }
     ]

**Comment by manisandro** [comment]
I've implemented the possibility to override the elevation interface directly in the viewer by assigning a elevation interface to `window.QWC2ElevationInterface`. An example for `api3.geo.admin.ch` is

```
const GeoAdminElevationInterface = {
    getElevation(pos, crs) {
        return new Promise((resolve, reject) => {
            const pos2056 = window.qwc2.CoordinatesUtils.reproject(pos, crs, "EPSG:2056");
            const params = {
                easting: pos2056[0],
                northing: pos2056[1],
                sr: 2056
            };
            const {axios} = window.qwc2.libs;
            axios.get("https://api3.geo.admin.ch/rest/services/height", {params}).then(response => {
                resolve(response.data.height);
            }).catch((e) => {
                reject(String(e));
            });
        });
    },
    getProfile(coordinates, distances, crs, samples) {
        return new Promise((resolve, reject) => {
            const coo2056 = coordinates.map(pos => window.qwc2.CoordinatesUtils.reproject(pos, crs, "EPSG:2056"));
            const params = {
                geom: JSON.stringify({type: "LineString", coordinates: coo2056}),
                sr: 2056,
                nb_points: samples
            };
            const {axios} = window.qwc2.libs;
            axios.get("https://api3.geo.admin.ch/rest/services/profile.json", {params}).then(response => {
                resolve(response.data.map(entry => entry.alts.DTM25));
            }).catch((e) => {
                reject(String(e));
            });
        });
    }
};
window.QWC2ElevationInterface = GeoAdminElevationInterface;
```

Save this as `assets/geoAdminElevation.js` and include it in your `index.html` 
```
<script type="text/javascript" src="assets/geoAdminElevation.js" ></script>
```
and you should be good!

**Note**: requires a `qwc-map-viewer:v2025.04.22` or newer.

**Comment by cymed** [comment]
Great! So in these cases the `qwc-elevation-service.json` becomes obsolete?

**Comment by manisandro** [comment]
Yes you can completely remove the elevation service if you decide to implement override the `QWC2ElevationInterface`.
